api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1574,8 +1574,82 @@ type EtcdSpec struct {
+	// unless overridden per-shard.
 	// +required
 	Storage ManagedEtcdStorageSpec `json:"storage"`
+
+	// shards configures etcd sharding by Kubernetes resource kind.
+	// When not specified, a default single shard accepting all prefixes is used.
+	// When specified, exactly one shard must have "/" in its resourcePrefixes.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:XValidation:rule="self.exists(s, '/' in s.resourcePrefixes)",message="exactly one shard must have '/' prefix"
+	// +kubebuilder:validation:XValidation:rule="self.all(s, s.resourcePrefixes.all(p, p == '/' || p.endsWith('#')))",message="non-default prefixes must end with '#'"
+	Shards []ManagedEtcdShardSpec `json:"shards,omitempty"`
+}
+
+// EtcdShardPriority defines the operational priority of an etcd shard
+// +kubebuilder:validation:Enum=Critical;High;Medium;Low
+type EtcdShardPriority string
+
+const (
+	EtcdShardPriorityCritical EtcdShardPriority = "Critical"
+	EtcdShardPriorityHigh     EtcdShardPriority = "High"
+	EtcdShardPriorityMedium   EtcdShardPriority = "Medium"
+	EtcdShardPriorityLow      EtcdShardPriority = "Low"
+)
+
+// ManagedEtcdShardSpec defines configuration for a single managed etcd shard
+type ManagedEtcdShardSpec struct {
+	// name is the unique identifier for this shard
+	// Must be DNS-1035 compliant (lowercase alphanumeric + hyphens)
+	// Used for resource naming: etcd-{name}, etcd-{name}-client, etc.
+	// +required
+	// +kubebuilder:validation:Pattern=`^[a-z]([-a-z0-9]*[a-z0-9])?$`
+	// +kubebuilder:validation:MaxLength=15
+	Name string `json:"name"`
+
+	// resourcePrefixes specifies which Kubernetes resources are stored in this shard
+	// Format: "group/resource#" or "/" for default (catch-all)
+	// Examples: "/events#", "/coordination.k8s.io/leases#", "/"
+	// Exactly one shard must have "/" as a prefix
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=50
+	// +kubebuilder:validation:items:MaxLength=255
+	// +listType=set
+	ResourcePrefixes []string `json:"resourcePrefixes"`
+
+	// priority determines operational importance and default backup frequency
+	// Critical: Default backup every 30 minutes
+	// High: Default backup hourly
+	// Medium/Low: Default backup disabled
+	// +optional
+	// +kubebuilder:default=Medium
+	Priority EtcdShardPriority `json:"priority,omitempty"`
+
+	// storage specifies storage configuration for this shard
+	// If not specified, inherits from ManagedEtcdSpec.Storage
+	// +optional
+	Storage *ManagedEtcdStorageSpec `json:"storage,omitempty"`
+
+	// replicas is the number of etcd replicas for this shard
+	// Must be 1 or 3. If not specified, defaults based on cluster's
+	// ControllerAvailabilityPolicy (1 for SingleReplica, 3 for HighlyAvailable)
+	// +optional
+	// +kubebuilder:validation:Enum=1;3
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Defaulting based on another resource is typically considered bad practice. I think this is what those utils were for, where/when is the defaulting called?